jax/experimental/ode.py

```diff
@@ -198,7 +203,7 @@ def body_fun(state):
     return carry, y_target
 
   f0 = func_(y0, ts[0])
-  dt = initial_step_size(func_, ts[0], y0, 4, rtol, atol, f0)
+  dt = initial_step_size(func_, ts[0], ts[1], y0, 4, rtol, atol, f0)
```

Comment:
The ODE solver can take more interpolation steps than output time steps. So I think it makes more to use the final time here:
```suggestion
  dt = initial_step_size(func_, ts[0], ts[-1], y0, 4, rtol, atol, f0)
```

Reply:
Ah, good call!